specs/agents/otel-distros.md

```diff
@@ -0,0 +1,60 @@
+# Building OpenTelemetry Distros
+
+> A distribution, not to be confused with a fork, is customized version of an OpenTelemetry component. A distribution is a wrapper around an 
+> upstream OpenTelemetry repository with some customizations.
+>
+> -- <cite>https://opentelemetry.io/docs/concepts/distributions/</cite>
+
+## Introduction
+
+Elastic distros are designed to support ease of use and enhanced features. We will leverage the underlying OpenTelemetry libraries for most functionality. Our thin wrapper over these libraries SHOULD support a familiar API. 
+
+Distributions should provide a simple, opinionated default configuration, listening to common sources (HTTP, SQL, etc.) and exporting data to an Elastic APM backend using OTLP. Additional features, not available in the OpenTelemetry library, can be introduced via appropriate extension points, with a view to contributing these upstream when possible.
+
+Distributions SHOULD follow this specification for behaviours and implemented functionality.
+
+## User Agent Headers
+
+Per the [OpenTelemetry Specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#user-agent), OpenTelemetry SDKs are expected to send a `User-Agent` header when exporting data to a backend. At a minimum, this header SHOULD identify the exporter, the language of its implementation, and the version of the exporter.
+
+Elastic distributions SHOULD configure a customized `User-Agent` header when possible. This allows data exported from a vanilla SDK and an Elastic distribution to be easily distinguished.
+
+To conform with [RFC7231](https://datatracker.ietf.org/doc/html/rfc7231#section-5.5.3), the exiting SDK `User-Agent` should be preceded by a product identifier and version for the distribution.
+
+```
+Elastic-Otel-Distro-<language>\<version> <original-sdk-user-agent>
```

Comment:
Is there a particular reason we could not use the `${telemetry.distro.name}-<language>\${telemetry.distro.version}` to have consistent values between the `user-agent` header and the resource attributes ?

If yes, then maybe moving the section about the user-agent might be better after properly defining the values for those attributes.

Reply:
The main reason I chose these in the first pass was to align with the convention the OpenTelemetry .NET SDK applies. It sends the `User-Agent` as OTel-OTLP-Exporter-Dotnet\<version>, while `telemetry.sdk.name` uses a simplified "open telemetry" value.

The casing of the proposed product name for the `User-Agent` is more in line with conventional casing and perhaps slightly more descriptive. The distro name won't include the fact it's a distro, for example, since it's redundant as the attribute name covers that part already.